app/controllers/storytime_admin/application_controller.rb

```diff
@@ -121,5 +123,24 @@ def current_user
         send("current_#{Admin.user_class_underscore_all}".to_sym)
       end
     end
+
+    def polymorphic_route_components(object_or_class_name)
+      object = nil
+
+      class_name = if object_or_class_name.is_a?(String)
+        object = object_or_class_name.constantize
+        object_or_class_name
+      elsif object_or_class_name.is_a?(Class)
+        object = object_or_class_name
+        object_or_class_name.name
+      else
```

Comment:
What would the object_or_class_name be if it got to this part of the conditional?